pkg/child/child.go

```diff
@@ -23,11 +23,25 @@ import (
 	"time"
 )
 
+// RunConfig are the inputs for a run operation.
+type RunConfig struct {
+	WorkingDir string
+	Command    string
+	Args       []string
+}
+
+// RunResult is the response from a run operation.
+type RunResult struct {
+	Stdout   []byte
+	Stderr   []byte
```

Comment:
Two questions:

1. Does it make sense to return bytes or string? 
2. Why not return an `io.Reader`?

The answer kinda depends on how the output is being used, but I'm a little bit worried that we're going to be loading mb (maybe even gb if debug logging is on) into memory.